### PR TITLE
Walking over people who are lying down with mechs hurts them

### PR DESCRIPTION
**Pull request by ghost**
[content]
Another idea by @drawsstuff 
Base damage is roughly 20-40 per treading, with a 5 second delay. Combat mechs do slightly higher damage at 30-60, (except for gygax and HONK, latter doing none at all) and with odysseus and clarkes doing none at all and marauders doing the highest at 40-80.

:cl:
 * rscadd: Mechs now do damage to mobs it treads over if they're lying down.

### CODE REVIEW

**Comment by Eneocho** [comment]
Mechs don't need any more buffs, they're already strong as fuck.

**Comment by adacovsk** [comment]
nerf phazon while you're at it, mechs are way too fast imo

**Comment by boy2mantwicethefam** [comment]
I like it

**Comment by SonixApache** [comment]
this is a super horrible idea

**Comment by GreatBigFailure** [comment]
Maybe just the super heavy slow ones

**Comment by Dacendeth** [comment]
Mechs need nerfs not buffs.

**Comment by gurfan** [comment]
This is a super good idea

**Comment by NiftyRyanyan** [comment]
Good for immersion, not for the gameplay.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -413,6 +417,26 @@
 	 playsound(src, get_sfx("mechstep"),40,1)
 	return result
 
+/obj/mecha/proc/StepOnCreature(var/mob/living/H)
+	if(step_over_cooldown && !H.lying) // Has to be on floor for this, mechs can't knock people over as easily.
```
Comment (Kammerjunk) [inline code/game/mecha/mecha.dm]: step_coolingdown

**Comment by Kammerjunk** [comment]
also to give some numbers, this does between ~~30 and 90~~ ~~20 and 60~~ 12 and 20 damage

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (ghost) [reply code/game/mecha/mecha.dm]: Fixed

**Comment by djkramer123** [comment]
since when do mechs need to be buffed

**Comment by githubuser4141** [comment]
I mean.. mechs in general have been heavily nerfed over the past year, and mechs were never as they are on TG. A mech here will be destroyed in 5-7 ion shots or 10-20 lasers, on TG they have higher armor and guns do a ton less damage, it's like 20-30 lasers or 10-15 ion rounds over there. And mining on /VG is way more tedious time consuming whereas on TG you can get enough materials minutes into the round with a single miner.

**Comment by djkramer123** [comment]
The only take away from that is that TG is, unsurprisingly, retarded. Mechs have been ""heavily"" nerfed and are still stupidly strong. There's no reason to increase their combat abilities.

**Comment by githubuser4141** [comment]
> The only take away from that is that TG is, unsurprisingly, retarded. Mechs have been ""heavily"" nerfed and are still stupidly strong. There's no reason to increase their combat abilities.

TBF on TG mechs DO get disadvantages like exit-time not being instant, stun + minor damage if the mech is broken when you're inside of it and take huge amounts of damage from attacks that hit their backside. I feel like VG mechs are all or nothing, e.g. you die instantly to someone with an ion gun or wipe the floor, on TG mech vs person fights are usually lengthier and either side can retreat if they want. 

This doesn't really change overall mech combat because if you're stunned a mech can just drill/punch you anyway, the latter being a thing ALL combat mechs can do which also stuns the punchee. So trying to walk over someone would actually be a waste of time in any combat mech as it does a lot less damage then just punching them. What it might do is piss off formerly neutral crew if your mech steps all over bodies they're trying to revive thus getting a lot of angry people after you.

**Comment by drawsstuff** [comment]
On second thought, it would be a far better solution that doesnt effect game balance if mechs just caused whoever they walked on to force scream and do some pain damage with a fluff message. If you *really* wanna make them do damage, have them break a random bone, but not stack brute. There shouldn't be a "new meta" of stun and run. I never thought about this when adding it to my code list.

**Comment by ghost** [comment]
Lack of support.

**Pull request closed without merging**